Babel macro support

morsdyce opened the pull request:
merge only after #18 

Please note that the import should be `reusable/macro` it is a rule of babel-macro-plugin.
In addition you can only import the reuse methods from the macro. You will get an error otherwise

morsdyce committed:
Add support for babel macro

morsdyce committed:
Fix reuseEffect not getting a debug name when index is 0

morsdyce committed:
Add files whitelist to npm publish

morsdyce committed:
Fix macro

morsdyce committed:
release alpha 5

adamkleingit commented on __tests__/reusable.macro.test.js:
```diff
@@ -0,0 +1,127 @@
+const pluginTester = require("babel-plugin-tester");
```
Why are the plugins unit tests here and not in the babel-plugin-tester repo?

adamkleingit commented on __tests__/reusable.macro.test.js:
```diff
@@ -0,0 +1,127 @@
+const reuseStateCode = `
+  import { reuseState } from "../src/reuseable.macro";
+  
+  const counter = () => {
```
Do you also support a case like this?
```
const counter = () => reuseState(0)
```

adamkleingit commented on __tests__/reusable.macro.test.js:
```diff
@@ -0,0 +1,127 @@
+  import { reuseMemo } from "../src/reuseable.macro";
+  
+  const counter = () => {
+    reuseMemo(() => {
```
Do you also support:
```
const counter = () => {
  const count = reuseMemo(() => ...)
}
```

morsdyce committed:
Change example to use published version

morsdyce committed:
Add additional use cases

morsdyce committed:
Update babel-plugin-reusable (uses real method name in fallback)

morsdyce commented:
Updated PR with suggested examples and changed behaviour to:
Unless a variable name is found, use a fallback based on parent unit and index as debugName

adamkleingit committed:
Merge branch 'master' into babel-macro
